ui/src/features/WorkspaceSettings/components/MembersSettings.tsx

```diff
@@ -105,26 +127,36 @@ const MembersSettings: React.FC = () => {
             </div>
           </div>
           <div className="max-h-[50vh] overflow-auto">
-            {members.map(member => (
-              <div key={member.userId} className="flex items-center gap-4 px-4 py-2">
-                <Checkbox
-                  className="border-zinc-600"
-                  checked={selectedMembers.includes(member.userId)}
-                  onClick={() =>
-                    setSelectedMembers(prev =>
-                      prev.includes(member.userId)
-                        ? [...prev.filter(pm => pm !== member.userId)]
-                        : [...prev, member.userId],
-                    )
-                  }
-                />
-                <p>{member.user.name}</p>
-                <p className="px-4 font-thin capitalize text-sm">{member.role}</p>
+            {members.map(m => (
+              <div key={m.userId} className="flex gap-4 px-4 py-2">
+                <p className="flex-1">{m.user?.name}</p>
+                <p className="flex-1 px-4 font-thin capitalize text-sm">{m.role}</p>
+                <DropdownMenu>
+                  <DropdownMenuTrigger className="flex-1 flex items-center gap-1">
+                    <p className="text-sm">{t("Change role")}</p>
+                    <CaretDown className="w-2 h-2" />
+                  </DropdownMenuTrigger>
+
+                  <DropdownMenuContent className="min-w-[70px]">
+                    {roles.map((role, idx) => (
+                      <DropdownMenuItem key={idx} onClick={() => handleChangeRole(m.userId, role)}>
+                        {role}
+                      </DropdownMenuItem>
+                    ))}
+                  </DropdownMenuContent>
+                </DropdownMenu>
+                <Button
+                  className="flex-1 h-[25px]"
+                  size="sm"
+                  variant="outline"
+                  onClick={() => handleRemoveMembers(m.userId)}>
+                  {t("Remove")}
+                </Button>
```

Comment:
I think lots will change in the coming months around roles as we update reearthx for more advanced role setting, but just to keep in your mind (if you have a good temp solution let me know) that we need to be careful who can do what action around this.
Basic musts are:
- non-Owners cannot change roles of others or remove them
- If its your personal workspace, you cannot change anything here
- etc

About the personal workspace, perhaps we just hide this route when its the currentWorkspace.

Reply:
Yes, can do that with personal workspace. We haven't started interated `role based access` on flow, once we have to clear set of actions a user can and can't do, we can build that into the UI.

Reply:
This is just something while testing we must avoid doing or else we will break workspace access